packages/model/src/custom/datetime.ts

```diff
@@ -1,38 +1,62 @@
-import { error, success } from '../result'
-import { CustomType, CustomTypeOpts, custom, string } from '../type-system'
+import { Result, error, success } from '../result'
+import { CustomType, OptionsOf, custom } from '../type-system'
+import { DecodingOptions } from 'src/decoder'
+import { ValidationOptions } from 'src/validate'
 
-export function datetime(opts?: CustomTypeOpts & { minimum?: Date; maximum?: Date }) {
-  return custom(
-    {
-      name: 'datetime',
-      format: 'date-time',
-      encodedType: string(),
-      decode: (input, options, decodeOptions) => {
-        let time: number = Date.parse(input)
-        if (Number.isNaN(time)) {
-          time = Number(input)
-          if (!decodeOptions?.cast || Number.isNaN(time) || time > 864000000000000 || time < -864000000000000) {
```

Comment:
why this check is no more needed? https://www.typescriptlang.org/play?#code/MYewdgziA2CmB00QHMAUZYHcAEARAhgC6yoAcAbACwAMtd9dAlIwLABQokMCSaGOBYmSoNR1AIzMgA

also this old check is wrong (it needs one more zero)

Reply:
It's because the date constructor already does the work of validating it for us: if the date is invalid the `.valueOf` the built date is a `NaN`
If you check the `tryMakeDate` function you'll see that it does return an error with an invalid date: [playground](https://www.typescriptlang.org/play?ssl=6&ssc=43&pln=1&pc=1#code/GYVwdgxgLglg9mABFATgTwLIEMDWBTAESyjwAoA3LAGxDwC5EwQBbAIzxUQB9EBnVGGADmASgZES3PgOGIA3gFgAUIkQQE-RABNieRAF5GeAO6IJZSjTwjlqlHiggUSAHIt2KAHQxeLrC9IdEk9LWgB5YFIREUQAfkQAIjA4KEQsbV0ExAYgvGUAX2VldTBeOCo8Tyo4IVJUTFxCXVIADgA2ABYABh7evp6ARmigA)

Reply:
ok cool!